langchain/vectorstores/redis.py

```diff
@@ -206,14 +209,24 @@ def add_texts(
         Returns:
             List[str]: List of ids added to the vectorstore
         """
-        ids = []
+
+        assert not (keys and ids), "`ids` and `keys` cannot be provided at the same time."
```

Comment:
lets raise a value error here (asserts only for internal logic/tests, not validating user input

Reply:
done. thanks!